File: backend/src/main/java/com/maeasoftworks/alfaconverterapi/dao/Log.java
package com.maeasoftworks.alfaconverterapi.dao;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Date;

@Entity
@Table(name = "logs")
@Getter
@Setter
public class Log {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(nullable = false)
    private long id;
    private final LocalDateTime timestamp;
    @Lob
    private final String conversion;
    private final int resultCode;

    public Log(LocalDateTime timestamp, String conversion, int resultCode) {
        this.conversion = conversion;
        this.resultCode = resultCode;
        this.timestamp = timestamp;
    }

    protected Log() {
        this.conversion = "";
        this.resultCode = -1;
        this.timestamp = LocalDateTime.now();
    }
}

File: backend/src/main/kotlin/com/maeasoftworks/alfaconverter/dao/Log.kt
package com.maeasoftworks.alfaconverter.dao

import com.maeasoftworks.alfaconverter.core.conversions.Conversion
import kotlinx.serialization.encodeToString
import kotlinx.serialization.json.Json
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "logs")
class Log(
	@Column
	val timestamp: LocalDateTime,
	@Lob
	@Column
	val conversion: String,
	@Column
	val resultCode: Int
) {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	@Column
	val id: Long = 0

	constructor(timestamp: LocalDateTime, conversion: Conversion, resultCode: Int) : this(timestamp, Json.encodeToString(conversion), resultCode)
}